js/components/main.js

```diff
@@ -482,37 +483,46 @@ class Main extends ImmutableComponent {
     currentWindow.on('maximize', function () {
       windowActions.setMaximizeState(true)
     })
-
     currentWindow.on('unmaximize', function () {
       windowActions.setMaximizeState(false)
     })
-
+    // NOTE: setTimeout is used to avoid spamming events
+    let resizeTimeout = null
     currentWindow.on('resize', function (event) {
-      windowActions.saveSize(event.sender.getSize())
+      if (resizeTimeout) {
+        clearTimeout(resizeTimeout)
+      }
+      resizeTimeout = setTimeout(function () {
+        const size = event.sender.getSize()
+        // NOTE: the default window size is whatever the last window resize was
+        appActions.defaultWindowParamsChanged(size)
+        windowActions.saveSize(size)
```

Comment:
I'm actually not sure if the window store receives app events, but if you need the window or session those are available in the browser process. DM me for details